src/Lykke.Quintessence.Rootstock.DomainServices/RootstockBlockchainService.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Threading.Tasks;
+using Lykke.Quintessence.Core.Blockchain;
+using Lykke.Quintessence.Core.Utils;
+using Lykke.Quintessence.Domain.Services.Strategies;
+using Lykke.Quintessence.RpcClient;
+using Newtonsoft.Json;
+
+namespace Lykke.Quintessence.Domain.Services
+{
+    public class RootstockBlockchainService : DefaultBlockchainService
+    {
+        private readonly IEthApiClient _ethApiClient;
+        
+        public RootstockBlockchainService(
+            IEthApiClient ethApiClient,
+            IDetectContractStrategy detectContractStrategy,
+            IGetTransactionReceiptsStrategy getTransactionReceiptsStrategy,
+            ITryGetTransactionErrorStrategy tryGetTransactionErrorStrategy,
+            Settings settings) 
+            
+            : base(ethApiClient, detectContractStrategy, getTransactionReceiptsStrategy, tryGetTransactionErrorStrategy, settings)
+        {
+            _ethApiClient = ethApiClient;
+        }
+
+        public override async Task<string> BroadcastTransactionAsync(
+            string signedTxData)
+        {
+            var serializedTransaction = signedTxData.HexToUTF8String();
+            var transaction = JsonConvert.DeserializeObject<DefaultRawTransaction>(serializedTransaction);
+
+            if (await _ethApiClient.GetTransactionAsync(transaction.Hash) != null)
+            {
+                return transaction.Hash;
+            }
+
+
+            var transactionHasBeenBroadcasted = false;
+            
+            for (var i = 0; i < 30; i++)
```

Comment:
Maybe polly could be used here for retires?

Reply:
I thought it is like common agreement

Reply:
Big nuget for single and very simple task seems over-engineering to me